obspy/clients/filesystem/tsindex.py

```diff
@@ -0,0 +1,1743 @@
+                                   "No leap seconds file will be used for "
+                                   "indexing.".format(file_path))
+            elif os.path.isfile(leap_seconds_file):
+                # use leap seconds file provided by user
+                leap_seconds_file = os.path.abspath(leap_seconds_file)
+            else:
+                raise OSError("No leap seconds file exists at `{}`. "
+                              .format(leap_seconds_file))
+            os.environ["LIBMSEED_LEAPSECOND_FILE"] = os.path.abspath(
+                                                        leap_seconds_file)
+        else:
+            # warn user and don't use a leap seconds file
+            logger.warning("No leap second file specified. This is highly "
+                           "recommended.")
+            os.environ["LIBMSEED_LEAPSECOND_FILE"] = "NONE"
+        return leap_seconds_file
+
+    def _is_index_cmd_installed(self):
+        """
+        Checks if the index command (e.g. mseedindex) is installed.
+
+        :raises OSError: If the index command is not installed.
+        """
+        try:
+            subprocess.call([self.index_cmd, "-V"])
+        except OSError:
+            raise OSError(
+                    "Required program `{}` is not installed. Hint: Install "
+                    "mseedindex at https://github.com/iris-edu/mseedindex/."
+                    .format(self.index_cmd))
+
+    @classmethod
+    def _run_index_command(cls, index_cmd, root_path, file_name, bulk_params):
+        """
+        Execute a command to perform indexing.
+
+        :type index_cmd: str
+        :param index_cmd: Name of indexing command to execute. Defaults to
+            `mseedindex`.
+        :type file_name: str
+        :param file_name: Name of file to index.
+        :type bulk_params: dict
+        :param bulk_params: Dictionary of options to pass to index_cmd.
+        """
+        try:
+            cmd = [index_cmd]
+            for option, value in bulk_params.items():
+                params = [option, value]
+                cmd.extend(params)
+            cmd.append(file_name)
+            # boolean options have a value of None
+            cmd = [c for c in cmd if c is not None]
+            proc = subprocess.Popen(cmd,
+                                    cwd=root_path,
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+            out, err = proc.communicate()
+            return (index_cmd, proc.returncode,
+                    out.strip(), err.strip())
+        except Exception as err:
+            msg = ("Error running command `{}` - {}"
+                   .format(index_cmd, err))
+            raise OSError(msg)
+
+
+class TSIndexDatabaseHandler(object):
+    """
+    Supports direct tsindex database data access and manipulation.
+    """
+
+    def __init__(self, sqlitedb, tsindex_table="tsindex",
```

Comment:
If there are at least vague plans that this gets abstracted to other SQL database flavors at some point, it would be better to rename `sqlitedb` to something more generic like `database` right away. Otherwise there will be painful `arg` renaming issues to deal with later on..